sysbuild/CMakeLists.txt

```diff
@@ -25,6 +25,31 @@ set(IMAGES)
 get_filename_component(APP_DIR  ${APP_DIR} ABSOLUTE)
 get_filename_component(app_name ${APP_DIR} NAME)
 
+# Propagate bootloader and signing settings from this system to the MCUboot and
+# application image build systems.
+if(SB_CONFIG_BOOTLOADER_MCUBOOT)
+  set(${app_name}_CONFIG_BOOTLOADER_MCUBOOT y CACHE STRING
+      "MCUBOOT is enabled as bootloader" FORCE
+  )
+  set(${app_name}_CONFIG_MCUBOOT_SIGNATURE_KEY_FILE
+      \"${SB_CONFIG_BOOT_SIGNATURE_KEY_FILE}\" y CACHE STRING
+      "Signature key file for signing" FORCE
+  )
+
+  # Set corresponding values in mcuboot
+  set(mcuboot_CONFIG_BOOT_SIGNATURE_TYPE_${SB_CONFIG_SIGNATURE_TYPE} y CACHE STRING
+      "MCUBOOT signature type" FORCE
+  )
+  set(mcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE
+      \"${SB_CONFIG_BOOT_SIGNATURE_KEY_FILE}\" y CACHE STRING
+      "Signature key file for signing" FORCE
+  )
+else()
+  set(${app_name}_CONFIG_BOOTLOADER_MCUBOOT n CACHE STRING
+      "MCUBOOT is disabled as bootloader" FORCE
+  )
+endif()
```

Comment:
actually those are placed here because those are settings that must be passed to images when MCUboot is enabled.

For example you should notice lines, such as:
https://github.com/zephyrproject-rtos/zephyr/blob/a7494fcfdb44d1ce3a324fe31fd90eaf5e8944c0/sysbuild/CMakeLists.txt#L31
https://github.com/zephyrproject-rtos/zephyr/blob/a7494fcfdb44d1ce3a324fe31fd90eaf5e8944c0/sysbuild/CMakeLists.txt#L34

so those are settings going to the main application, and must be set before calling:
https://github.com/zephyrproject-rtos/zephyr/blob/a7494fcfdb44d1ce3a324fe31fd90eaf5e8944c0/sysbuild/CMakeLists.txt#L54-L58

thus, moving this to `bootloader/CMakeLists.txt` will be to late.

Let me clean up this, so that handling of Kconfig settings which must be passed from sysbuild to other build systems becomes cleaner.